### ISSUE TITLE
Scrolling through streams causes Homebridge to terminate

### ISSUE THREAD

**Issue by ceeeekay** [dgreif/ring#562]
# Bug Report

### Describe the Bug
 When scrolling though multiple cameras, e.g., in ATV Control Center, if a stream starts after it's no longer required, a SIGTERM is generated for Homebridge. At this point cameras are unavailable and show a broken stream icon while Homebridge restarts.
Methodically going to each camera works fine, but scrolling quickly past other cameras, e.g., four cameras in this case, causes the error. 

### To Reproduce
 - Open Control Center on ATV (long press TV button)
 - Open Home icon
 - Scroll sideways past multiple cameras.

### Expected behavior
 Cameras appear normal, and their streams can be viewed either in preview or full screen.

### Screenshots/Logs
 Homebridge log:
```
[1/29/2021, 4:06:47 PM] [Ring] Preparing Live Stream for Deck
[1/29/2021, 4:06:48 PM] [Ring] Waiting for stream data from Deck (1.092s)
[1/29/2021, 4:06:49 PM] [Ring] Received stream data from Deck (1.545s)
[1/29/2021, 4:06:49 PM] [Ring] Streaming active for Deck
[1/29/2021, 4:06:49 PM] [Ring] Stopped Live Stream for Deck
[1/29/2021, 4:06:49 PM] [Ring] Preparing Live Stream for Driveway
[1/29/2021, 4:06:50 PM] [Ring] Waiting for stream data from Driveway (1.078s)
[1/29/2021, 4:06:50 PM] [Ring] Received stream data from Driveway (1.29s)
[1/29/2021, 4:06:50 PM] [Ring] Streaming active for Driveway
[1/29/2021, 4:06:50 PM] [Ring] Stopped Live Stream for Driveway
Error: SIP request made after call was destroyed
    at SipCall.request (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:117:35)
    at SipCall.sendKeyFrameRequest (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:188:21)
    at SipCall.<anonymous> (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:216:24)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:5:58)
Error: SIP request made after call was destroyed
    at SipCall.request (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:117:35)
    at SipCall.sendKeyFrameRequest (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:188:21)
    at SipCall.<anonymous> (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:216:24)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:5:58)
[1/29/2021, 4:06:51 PM] Got SIGTERM, shutting down Homebridge...
Error: SIP request made after call was destroyed
    at SipCall.request (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:117:35)
    at SipCall.sendKeyFrameRequest (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:188:21)
    at SipCall.<anonymous> (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:216:24)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:5:58)
Error: SIP request made after call was destroyed
    at SipCall.request (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:117:35)
    at SipCall.sendKeyFrameRequest (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:188:21)
    at SipCall.<anonymous> (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:216:24)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/lib/node_modules/homebridge-ring/lib/api/sip-call.js:5:58)
[1/29/2021, 4:06:56 PM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[1/29/2021, 4:07:01 PM] [HB Supervisor] Restarting Homebridge...
```

### Additional context
 N/A

### Homebridge Ring Config
```
        {
            "refreshToken": "......",
            "sendCameraMotionNotificationsToTv": true,
            "sendDoorbellMotionNotificationsToTv": false,
            "platform": "Ring"
        },
```

### Environment

 - Ring plugin: v9.14.1
 - OS: Ubuntu 19.10 / eoan
 - Node.js: v12.18.3
 - NPM: 6.14.6
 - homebridge:1.2.5
 - hoobs: N/A

**Comment by ceeeekay** [comment]
Just a note: I should probably rebuild this with 20.04 LTS.  If that's the cause we can probably just close this.

**Comment by ceeeekay** [comment]
Hi, I've just rebuilt with a fresh 20.04 and this is still happening.

### Environment
 - Ring plugin: 9.14.2 
 - OS: Ubuntu 20.04
 - Node.js: 12.20.1 
 - NPM: 6.14.10 
 - homebridge:1.1.7
 - hoobs: N/A

**Comment by dgreif** [comment]
Thanks, I'll take a look when I get a chance. Most users aren't switching between camera streams quickly, but if it's possible on Apple TV, then it's something that shouldn't break the plugin, and especially the whole homebridge process.

**Comment by ceeeekay** [comment]
Yeah unfortunately when selecting a stream in ATV / Home your only option is to scroll through them, which opens & closes streams very quickly. Cheers.

**Comment by tsightler** [comment]
I think I'm seeing this behavior in a weird way as well.  I was trying to hack a workaround for the lack of ability to get a snapshot from battery cameras, by grabbing a 1 second video and grabbing a frame to use for the snapshot image.  In my prototype, this worked fine, but when I tried to implement it in the real code, I kept getting this error about 95% of the time.  If I changed to record 5 seconds, no error.

Then I tried just using recordToFile() with 1 second, and, while it worked most of the time, even there it would fail sometimes.  With 5 seconds, it always seems to work.

So it seems like really short SIP sessions trigger this behavior regardless.  Admittedly, it's not really a big deal as I was really abusing this somewhat, it was just a crazy idea to keep me from having so many support request about missing snapshots on battery cameras and actually it still works, I just have to run the stream for 5 seconds instead of 1.
